controllers/helmchart_controller.go

```diff
@@ -1090,3 +1141,22 @@ func reasonForBuild(build *chart.Build) string {
 	}
 	return sourcev1.ChartPullSucceededReason
 }
+
+func chartRepoErrorReturn(err error, obj *sourcev1.HelmChart) (sreconcile.Result, error) {
```

Comment:
Based on the above issues with the usage of this function, I think it'd be better to rename this function to make it explicit that it'll return a stalling error and must be used only when we need a user interaction.
Maybe `chartRepoConfigErrorReturn` would be more appropriate as it emphasizes on config error which needs user intervention to fix.
Also, a comment with warning would also be useful to help avoid its unintended usage. Stalling errors result in no reconciliation, so we need to be extra careful when using it.

Reply:
see https://github.com/fluxcd/source-controller/pull/728